File: hackerrank/problem solving/algorithms/1 - warmup/7 - staircase/staircase.java
// https://www.hackerrank.com/challenges/staircase/problem?isFullScreen=true

void main() {
    int n = Integer.parseInt(IO.readln());
    staircase(n);
}

// T: O(n^2)
// S: O(n) extra space
void staircase(int n) {
    for (int i = 1; i <= n; i++) {
        var spaces = " ".repeat(n - i);
        var hashes = "#".repeat(i);
        IO.println(spaces + hashes);
    }
}

File: hackerrank/problem solving/algorithms/1 - warmup/7 - staircase/staircase.go
// https://www.hackerrank.com/challenges/staircase/problem?isFullScreen=true

package main

import (
    "fmt"
    "strings"
)

func main() {
    var n int
    _, _ = fmt.Scan(&n)
    staircase(n)
}

// T: O(n^2)
// S: O(n) extra space
func staircase(n int) {
    for i := 1; i <= n; i++ {
        spaces := strings.Repeat(" ", n-i)
        hashes := strings.Repeat("#", i)
        fmt.Println(spaces + hashes)
    }
}
